sdk/core/core-rest-pipeline/src/policies/formDataPolicy.ts

```diff
@@ -46,39 +53,53 @@ function wwwFormUrlEncode(formData: FormDataMap): string {
 }
 
 async function prepareFormData(formData: FormDataMap, request: PipelineRequest): Promise<void> {
-  const requestForm = new FormData();
-  for (const formKey of Object.keys(formData)) {
-    const formValue = formData[formKey];
-    if (Array.isArray(formValue)) {
-      for (const subValue of formValue) {
-        requestForm.append(formKey, subValue);
-      }
-    } else {
-      requestForm.append(formKey, formValue);
-    }
+  if (request.body) {
+    throw new Error("multipart/form-data request must not have a request body already specified");
   }
 
-  request.body = requestForm;
+  // clear request.formData
   request.formData = undefined;
+
+  // validate content type (multipart/form-data)
   const contentType = request.headers.get("Content-Type");
-  if (contentType && contentType.indexOf("multipart/form-data") !== -1) {
-    request.headers.set(
-      "Content-Type",
-      `multipart/form-data; boundary=${requestForm.getBoundary()}`
-    );
+  if (contentType && !contentType.startsWith("multipart/form-data")) {
+    // content type is specified and is not multipart/form-data. Exit.
+    return;
   }
-  try {
-    const contentLength = await new Promise<number>((resolve, reject) => {
-      requestForm.getLength((err, length) => {
-        if (err) {
-          reject(err);
-        } else {
-          resolve(length);
+
+  request.headers.set("Content-Type", contentType ?? "multipart/form-data");
+
+  // set body to MultipartRequestBody using content from FormDataMap
+  const parts: BodyPart[] = [];
+
+  for (const [fieldName, values] of Object.entries(formData)) {
+    for (const value of Array.isArray(values) ? values : [values]) {
+      if (typeof value === "string") {
+        parts.push({
+          headers: createHttpHeaders({
+            "Content-Disposition": `form-data; name="${fieldName}"`,
+          }),
+          body: stringToUint8Array(value, "utf-8"),
+        });
+      } else {
+        // using || instead of ?? here since if value.name is empty we should create a file name
+        const fileName = value.name || "blob";
```

Comment:
Is setting a default filename permitted by the spec? I am mainly worried whether could this be interpreted unexpectedly somehow.

Reply:
I added this to replicate the behavior of Chrome's FormData: if no filename is specified (or it's an empty string) the name 'blob' is used.

Reply:
Well then no objection in this case, thanks for clarifying!